### DISCUSSION TITLE
Remove UserId from JokeCreateDto

### DISCUSSION THREAD

**Discussion by heijmn** [Code2Gether-Discord/JokesOnYou#145] [Ideas]
Currently the JokeCreateDto class contains a UserId property which shouldn't be set by the client API. The server ignores the value anyhow and set's the value itself. So it shouldn't be part of the Dto in the first place.

A UserId is however required to create a valid Joke in the system, so it needs to be retrieved and passed on somewhere. My suggestion would be to keep it simple and make the service responsible for combining the required data and thus provide the data as separate parameters (JokeCreateDto and User / UserId).

**Comment by HellHunterMax** [comment]
Yes I do agree, I didn't like this way too.
My first idea was to add parameters too.
We changed it to a more all encompassing Dto.
@chrisK00 (I think) suggested we use different Dto's for different parts of the system, which could be a good solution too.

**Reply by chrisK00** [reply]
1. Yes i suggested having 3 different models (which makes sense if you want to follow n-tier architecture). 

Since MediatR seems to be pretty much a standard today, the option above is the most common. The only reason not to implement this is that we are trying to keep things as simple as possible, this would include more code and what some might consider duplicated in this case. 

2. But passing the userId as a parameter from what ive been told by other devs is not a good idea since the request model - should have everything the request requires. 

Tl:dr if we are gonna make this change, i would suggest option 1 or just keep it as is. I mean after all our user doesnt actually know he can send in the user id, our angular client takes care of that and we are the ones developing it.
